azure-pipelines/builds/ci.yml

```diff
@@ -15,24 +15,69 @@ variables:
   - name: Codeql.Enabled
     value: true
 
+  - name: defaultPoolName
+    ${{ if eq(variables['System.TeamProject'], 'public') }}:
+      value: $[replace(replace(eq(contains(coalesce(variables['System.PullRequest.TargetBranch'], variables['Build.SourceBranch'], 'refs/heads/main'), 'release'), 'true'), True, 'NetCore-Svc-Public' ), False, 'NetCore-Public')]
+    ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+      value: $[replace(replace(eq(contains(coalesce(variables['System.PullRequest.TargetBranch'], variables['Build.SourceBranch'], 'refs/heads/main'), 'release'), 'true'), True, 'NetCore1ESPool-Svc-Internal'), False, 'NetCore1ESPool-Internal')]
```

Comment:
That is already handled by arcade: https://github.com/dotnet/arcade/blob/main/eng/common/templates/variables/pool-providers.yml

Include the variables file like this: https://github.com/dotnet/aspire/blob/74c40701ec6385bcccc2452863ffb3585d36601e/eng/pipelines/azure-pipelines.yml#L18

Reply:
(I am unable to reopen this merged PR. I have linked a [new PR with these changes](https://github.com/dotnet/source-build-reference-packages/pull/830/files#diff-33ae5736679b1efe12fde9f3fb58dbe360dd8a8bb32e5b98620bf7d644114f19R15).)

Good to know!

The reason I defined the values in this manner is because [`eng/common/templates/job/source-build.yml`](https://github.com/dotnet/source-build-reference-packages/blob/main/eng/common/templates/job/source-build.yml#L50) defines the values in this same way.

I'm surprised that `eng/common/templates/job/source-build.yml` doesn't also make use of these variables.

Reply:
The source-build template existed prior to the pool-providers.yml.  It wasn't refactored to make use of it once it was defined.